app/api/ada/index.js

```diff
@@ -1135,6 +1135,87 @@ export default class AdaApi {
     }
   }
 
+  /**
+   * Restore all addresses like restoreWallet() but do not touch storage.
+   * TBD: this function is based on restoreWallet() and duplicate parts of it.
+   */
+  async restoreWalletForTransfer(
+    recoveryPhrase: string,
+    checkAddressesInUse: FilterFunc
+  ): Promise<Wallet> {
+    Logger.debug('AdaApi::restoreWalletForTransfer called');
+    const walletName = '';
+    const walletPassword = '';
+    const assurance = 'CWANormal';
+    const unit = 0;
+
+    const walletInitData = {
+      cwInitMeta: {
+        cwName: walletName,
+        cwAssurance: assurance,
+        cwUnit: unit
+      },
+      cwBackupPhrase: {
+        bpToList: recoveryPhrase // array of mnemonic words
+      }
+    };
+
+    try {
+      // recover master key
+      const adaWallet = toAdaWallet(walletInitData);
+      const masterKey = generateWalletMasterKey(
+        walletInitData.cwBackupPhrase.bpToList,
+        walletPassword
+      );
+      // always create the 0th account
+      const cryptoAccount = createCryptoAccount(masterKey, walletPassword, 0);
+
+      const localAccount = {
+        account: cryptoAccount.account,
+        root_cached_key: cryptoAccount.root_cached_key.key().to_hex(),
+        derivation_scheme: cryptoAccount.derivation_scheme
+      };
+
+      const _saveAsAdaAddresses = async _request => {
```

Comment:
You will actually need these addresses as part of the return time to build the transfer function since you will need to fetch all the UTXO for these addresses. Maybe save these as an array of address strings and add it to the function return type?

Reply:
This is what I've thought. But since I haven't been into that part I was a bit unsure. I will change this.